numpy/random/tests/test_random.py

```diff
@@ -205,12 +208,13 @@ def test_repeatability(self):
         assert_(tgt[np.dtype(np.bool).name] == res)
 
 
-class TestRandomDist(TestCase):
+class TestRandomDist:
     # Make sure the random distribution returns the correct value for a
     # given seed
 
-    def setUp(self):
-        self.seed = 1234567890
+    @classmethod
```

Comment:
I share @matthew-brett and @charris's vague dislike of generator-based tests. It's a nice idea and does work, but just a bit too clunky in practice.